nav2_costmap_2d/plugins/static_layer.cpp

```diff
@@ -458,6 +467,11 @@ StaticLayer::updateCosts(
       }
     }
   }
+
+  // restore the map region occupied by the polygon using cached data
+  for (auto [cell, cost_value] : map_region_to_restore) {
```

Comment:
I think you can make a costmap_2d method for `setCellsOccupiedByPolygon` instead - which might be nice to use in other places anyway :-)